File: bad-tests-v2/kernels_construct_firstprivate_clause.cpp
#include "acc_testsuite.h"

#ifndef T1
//T1:kernels construct firstprivate clause,V:2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);

    int* a = (int*)malloc(sizeof(int) * 10);
    int* b = (int*)malloc(sizeof(int) * 10);
    int* c = (int*)malloc(sizeof(int) * 10);

    for (int i = 0; i < 10; i++) {
        a[i] = i;
        b[i] = i;
        c[i] = i;
    }

    #pragma acc kernels firstprivate(a, b, c)
    {
        for (int i = 0; i < 10; i++) {
            a[i] = b[i] + c[i];
        }
    }

    for (int i = 0; i < 10; i++) {
        if (a[i] != b[i] + c[i]) {
            err = 1;
            break;
        }
    }

    free(a);
    free(b);
    free(c);

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: Phind-CodeLlama-34B-v2/context-template-full/kernels_construct_firstprivate_clause.f90
#ifndef T1
!T1:kernels construct firstprivate clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER :: a = 10
        INTEGER :: b = 20
        INTEGER :: c = 30
        INTEGER :: d = 40
        INTEGER, DIMENSION(LOOPCOUNT) :: hostArray, hostArray2
        INTEGER, DIMENSION(LOOPCOUNT) :: deviceArray, deviceArray2

        hostArray = 5
        hostArray2 = 10

        !$acc kernels firstprivate(a, b)
          deviceArray = hostArray + a
          deviceArray2 = hostArray2 + b
        !$acc end kernels

        DO i = 1, LOOPCOUNT
          IF (deviceArray(i) .ne. (hostArray(i) + a)) THEN
            errors = errors + 1
          END IF
          IF (deviceArray2(i) .ne. (hostArray2(i) + b)) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM